New dimensiondata vlan

tintoy opened the pull request:
##### ISSUE TYPE
- Feature Pull Request
##### COMPONENT NAME

ansible-module-extras/cloud/dimensiondata/dimensiondata_vlan.py
##### ANSIBLE VERSION

```
ansible 2.2.0
```
##### SUMMARY

This is Ansible module support for apache-libcloud functions.
Specifically, this code supports Ansible invocations of DimensionData's 
VLAN functionality.

This is to support a new capability within ansible, so there is no "BEFORE".
Note that this routine has an "action verb" which allows multiple different
actions to be performed on a vlan.  Below shows the ansible call to
create a vlan, along with its output.

```
    - name: create vlan named "{{vlan_name}}"
       dimensiondata_vlan:
         region: "{{dimension_data_region}}"
         location: "{{dimension_data_location}}"
         network_domain: "{{dimension_data_network_domain}}"
         name: "{{vlan_name}}"
         description: A test VLAN Network.
         private_ipv4_base_address: "{{base_ip}}"
         private_ipv4_prefix_size: "{{prefix_size|int}}"
         action: create
         wait: true
         wait_time: 300

    TASK [create vlan named "demo-vlan"] *******************************************
    changed: [127.0.0.1] => {"changed": true, "msg": "Successfully created VLAN.", "vlan": {"description": "A test VLAN Network.", "id": "9755b966-b07a-418a-a892-de5416ee992f", "location": "NA12", "name": "demo-vlan", "private_ipv4_range_address": "192.168.24.0", "private_ipv4_range_size": 24, "status": "NORMAL"}}

```
###### REQUIRES

https://github.com/ansible/ansible/pull/17604
https://github.com/apache/libcloud/pull/858

Commit:
Add dimensiondata_vlan module

jr2730 committed:
fixed error formatting

tintoy commented:
Migrated from https://github.com/ansible/ansible-modules-extras/pull/2977 by tintoy (not original author)

tintoy commented:
Rebasing against `ansible/devel`.

tintoy committed:
Tidy up code based on previous PR feedback from @abadger.

tintoy commented:
Ok, this should be ready for review (have made the same changes as requested in #19325).

mattclay commented:
@tintoy CI failures are due to the following:

```
2017-01-21 08:36:07 ============================================================================
2017-01-21 08:36:07 lib/ansible/modules/cloud/dimensiondata/dimensiondata_vlan.py
2017-01-21 08:36:07 ============================================================================
2017-01-21 08:36:07 ERROR: Type comparison using type() found on line 263. Use isinstance() instead
2017-01-21 08:36:07 ERROR: Type comparison using type() found on line 299. Use isinstance() instead
2017-01-21 08:36:07 ERROR: Type comparison using type() found on line 380. Use isinstance() instead
2017-01-21 08:36:07 ERROR: Type comparison using type() found on line 411. Use isinstance() instead
```

tintoy commented:
Ah, I'll look into that now.

tintoy committed:
Use isinstance instead of type comparison (ansible/ansible#20321).

tintoy commented:
Done :)

tintoy committed:
Changes based on feedback from ansible/ansible#20583.

tintoy committed:
Further changes based on feedback from ansible/ansible#20583

- Documentation for mcp_user and mcp_password module parameters.
- Use of C(xxx) in documentation.

tintoy committed:
Enable dimensiondata_vlan module to accept either Id or name for target network domain.
Fix broken import of ConfigParser in module_utils/dimensiondata.py.
Also, fix buggy detection of MCP 2.0 datacenters (really, it should not be using the name, but that's a libcloud issue).

tintoy committed:
Update change-log to note new dimensiondata_network and dimensiondata_vlan modules.

tintoy commented:
Actually, I have reviewed this module and am not satisfied with its design (unlike our other modules it is neither declarative nor idempotent). So I will close this PR, and open a new one once the new module is ready.
